tests/integration/teal/annotated/AlgoBank_h0_c1.tealf

```diff
@@ -0,0 +1,226 @@
+assert                                 //  (293)  Assert(payment.get().receiver() == Global.current_application_address())
+load 6                                 //  (294)  sender.address()
+txnas Accounts                         //  (296)
+byte "balance"                         //  (298)  Bytes('balance')
+load 6                                 //  (299)  sender.address()
+txnas Accounts                         //  (301)
+byte "balance"                         //  (303)  Bytes('balance')
+app_local_get                          //  (304)  App.localGet(sender.address(), Bytes('balance'))
+load 5                                 //  (305)  payment.get()
+gtxns Amount                           //  (307)  payment.get().amount()
++                                      //  (309)  App.localGet(sender.address(), Bytes('balance')) + payment.get().amount()
+app_local_put                          //  (310)  App.localPut(sender.address(), Bytes('balance'), App.localGet(sender.address(), Bytes('balance')) + payment.get().amount())
+retsub                                 //  (311)  def deposit(payment: abi.PaymentTransaction, sender: abi.Account) -> Expr:
+                                       //         def getBalance(user: abi.Account, *, output: abi.Uint64) -> Expr:
+// getBalance                          //
+getBalance_2:                          //
+txnas Accounts                         //  (312)  user.address()
+byte "balance"                         //  (314)  Bytes('balance')
+app_local_get                          //  (315)  App.localGet(user.address(), Bytes('balance'))
+retsub                                 //  (316)  def getBalance(user: abi.Account, *, output: abi.Uint64) -> Expr:
+                                       //         def withdraw(amount: abi.Uint64, recipient: abi.Account) -> Expr:
+// withdraw                            //
+withdraw_3:                            //
+store 8                                //  (317)  {COMPILE}
+store 7                                //  (319)
+txn Sender                             //  (321)  Txn.sender()
+byte "balance"                         //  (323)  Bytes('balance')
+txn Sender                             //  (324)  Txn.sender()
+byte "balance"                         //  (326)  Bytes('balance')
+app_local_get                          //  (327)  App.localGet(Txn.sender(), Bytes('balance'))
+load 7                                 //  (328)  amount.get()
+-                                      //  (330)  App.localGet(Txn.sender(), Bytes('balance')) - amount.get()
+app_local_put                          //  (331)  App.localPut(Txn.sender(), Bytes('balance'), App.localGet(Txn.sender(), Bytes('balance')) - amount.get())
+itxn_begin                             //  (332)  InnerTxnBuilder.Begin()
+int pay                                //  (333)  def deposit(payment: abi.PaymentTransaction, sender: abi.Account) -> Expr:
```

Comment:
This is a new bug!!!

Reply:
I'm trying to assess the impact on other contracts. I might end up reverting.

Reply:
This is the only such bug I've identified. I'm proceeding with caution with more improvements to the router.

Reply:
This bug and other improvements for source mapping the router is addressed in #676 .  I want to keep the review process for this PR as straight forward as possible, so I'll incorporate suggested improvements for code organization, performance, api, etc. here. However, suggestions that impact the actual source mapping results would be incorporated into #676.